src/App.vue

```diff
@@ -801,19 +801,19 @@ export default class App extends Mixins(CommonMixin, DateMixin, FilingTemplateMi
       const paymentToken = filingComplete?.header?.paymentToken
       if (paymentToken) {
         const isPaymentActionRequired: boolean = filingComplete.header?.isPaymentActionRequired
-        const dashboardUrl = sessionStorage.getItem('DASHBOARD_URL')
+        const returnUrl = sessionStorage.getItem('DASHBOARD_URL') + this.getBusinessId +
+          `?filing_id=${this.getFilingId}`
 
         // if payment action is required, navigate to Pay URL
         if (isPaymentActionRequired) {
           const authUrl = sessionStorage.getItem('AUTH_WEB_URL')
-          const returnUrl = encodeURIComponent(dashboardUrl + this.getBusinessId)
-          const payUrl = authUrl + 'makepayment/' + paymentToken + '/' + returnUrl
+          const payUrl = authUrl + 'makepayment/' + paymentToken + '/' + encodeURIComponent(returnUrl)
           // assume Pay URL is always reachable
           // otherwise user will have to retry payment later
           navigate(payUrl)
         } else {
-          // navigate to Dashboard URL
-          navigate(dashboardUrl + this.getBusinessId)
+          // otherwise go straight to dashboard
+          navigate(returnUrl)
```

Comment:
This chunk of code is used for non-correction filings (which use the newer Fee Summary component with built-in actions).